### PR TITLE
Bug fix for upgrade GUID change

### PR DESCRIPTION
**Pull request by uiri**
Reduce log spam from skipping an upgrade based on the GUID staying the same.

Ensure that the Agent reports the correct extension version and its state as enabled in cases where the GUID is the same.

Fix unit testing for the upgrade GUID feature

### CODE REVIEW

File: azurelinuxagent/ga/exthandlers.py [hunk 1]
```diff
@@ -223,7 +223,15 @@ def run_status(self):
         return
 
     def get_guid(self, name):
-        return self.last_guids.get(name, None)
+        return self.last_guids.get(name, (None, False))[0]
```
Comment (hglkrijger) [inline azurelinuxagent/ga/exthandlers.py]: should you check for None before reference? also the method below

File: azurelinuxagent/ga/exthandlers.py [hunk 2]
```diff
@@ -223,7 +223,15 @@ def run_status(self):
...
+
+    def get_log_guid(self, ext_handler):
```
Comment (hglkrijger) [inline azurelinuxagent/ga/exthandlers.py]: consider renaming `log_guid`, I think it is a bit misleading

File: azurelinuxagent/ga/exthandlers.py [hunk 3]
```diff
@@ -223,7 +223,15 @@ def run_status(self):
...
+        return self.last_guids.get(ext_handler.name, (None, False))[1]
+
+    def set_log_guid(self, ext_handler, log_val):
+        guid = self.get_guid(ext_handler.name)
+        if guid is not None:
+            self.last_guids[ext_handler.name] = (guid, log_val)
```
Comment (hglkrijger) [inline azurelinuxagent/ga/exthandlers.py]: Consider renaming `last_guids` as well, the context is not obvious to the reader

File: azurelinuxagent/ga/exthandlers.py (hunk 2 above)
Comment (uiri) [reply azurelinuxagent/ga/exthandlers.py]: Fixed.

File: azurelinuxagent/ga/exthandlers.py (hunk 3 above)
Comment (uiri) [reply azurelinuxagent/ga/exthandlers.py]: fixed.

File: azurelinuxagent/ga/exthandlers.py (hunk 1 above)
Comment (uiri) [reply azurelinuxagent/ga/exthandlers.py]: `last_upgrade_guids` is initialized to an empty dictionary when the ExtHandlerHandler is initialized.
The second parameter to `get` specifies what is returned if `name` is not found in the dictionary.

**Pull request merged**